Extensions/vanilla_video.js

```diff
@@ -54,15 +54,37 @@ XKit.extensions.vanilla_video = {
 	},
 
 	running: false,
+
 	run: function() {
 		this.running = true;
+		this.observer = new MutationObserver(this.add_controls);
+
 		if (this.preferences.background_color.value === "") {
 			this.preferences.background_color.value = "#000000";
 		}
+
+		if (XKit.page.react) {
+			XKit.tools.add_css("video + div {display: none !important;}");
+			this.observer.observe(document.body, {childList: true, subtree: true});
```

Comment:
this is going to be super expensive and laggy unfortunately :( is there a better way we can do this?

Reply:
not that i know of, since the entire `timeline` element gets replaced/added just... whenever, and we want to listen for all new posts (or in this case, videos)

the only thing the post listener does differently is actually check through the individual mutations before activating, but still has to observe the entire `document.body` for the same reason

Reply:
and i mean we could make one MutationObserver to observe the body and then apply more specific MutationObservers when it finds a new timeline to listen to, but then we're still observing the entire body and checking for specific elements every single time there's a mutation.

Reply:
(for reference, what the post listener *used to* do was count how many post elements there were on a 1-second loop, and activate if it found more than the last time it counted, and we didn't get any performance complaints when we switched it to MutationObserver)